File: A. Easy/1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* replaceElements(int* arr, int arrSize, int* returnSize) {
    *returnSize = arrSize;
    int* result = (int*)malloc(arrSize * sizeof(int));
    int largest = -1;
    for (int i = arrSize - 1; i >= 0; i--) {
        result[i] = largest;
        if (arr[i] > largest) largest = arr[i];
    }
    return result;
}

File: 2. Arrays 101/1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        largest = -1
        for i in range(len(arr) - 1, -1, -1):
            if (arr[i] > largest):
                largest, arr[i] = arr[i], largest
            else:
                arr[i] = largest
        return arr


        